modules/ardrivo/src/Wire.cxx

```diff
@@ -0,0 +1,142 @@
+#include "Wire.h"
+#include "Arduino.h"
+#include "BoardDataDef.hxx"
+
+void TwoWire::begin()
+{
+    begun = true;
+}
+
+void TwoWire::begin(uint8_t address)
+{
+    slaveAddress = address;
+    begun = true;
+}
+void TwoWire::begin(int address)
+{
+    slaveAddress = address;
+    begun = true;
+}
+
+// requestFrom = REQUEST_SEND
+size_t TwoWire::requestFrom(uint8_t address, size_t size, bool stop)
+{
+ 
+    ProtocolRequest req = ProtocolRequest{ REQUEST_SEND, size };
+    transmissionAddress = address;
+    board_data->i2c_buses[bus_id].requests[address].push_back(req);
+
+    size_t bytesReturned = req.bytes;
+    return bytesReturned;
+}
+
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size)
+{
+    return requestFrom(address, static_cast<size_t>(size), true);
+}
+
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size, uint8_t stop)
+{
+    return requestFrom(address, static_cast<size_t>(size), static_cast<bool>(stop));
+}
+
+uint8_t TwoWire::requestFrom(int address, int size)
+{
+    return requestFrom(static_cast<uint8_t>(address), static_cast<size_t>(size), true);
+}
+
+uint8_t TwoWire::requestFrom(int address, int size, int stop)
+{
+    return requestFrom(static_cast<uint8_t>(address), static_cast<size_t>(size),static_cast<bool>(stop));
+}
+
+void TwoWire::beginTransmission(uint8_t adress)
+{
+    txBufferIndex = 0;
+    // set address of targeted slave    
+    transmissionAddress = adress;
+}
+void TwoWire::beginTransmission(int adress)
+{   
+    beginTransmission((uint8_t)(adress));  
+}
+
+uint8_t TwoWire::endTransmission()
+{
+    return endTransmission(true); 
+}
+uint8_t TwoWire::endTransmission(uint8_t)
+{
+    if (transmitted > 0)
+    {
+        transmissionAddress.reset();
+        txBufferIndex = 0;
+        //0 = sucess transmission
+        return 0;
+    }
+    transmissionAddress.reset();
+    //Error at transmission
+    return 1;
+};
+
+// REQUEST_RECEIVE
+size_t TwoWire::write(uint8_t value)
+{
+    if (transmissionAddress)
+    {       
+        ProtocolRequest req = ProtocolRequest{ REQUEST_RECEIVE, value };
+        board_data->i2c_buses[bus_id].requests[transmissionAddress.value()].push_back(req);
+
+        // put byte in tx buffer
+        board_data->i2c_buses[transmissionAddress.value()].tx[txBufferIndex] = ((std::byte)value); 
+        txBufferIndex++;
+        transmitted = value;
+        // update amount in buffer   
+        board_data->i2c_buses[transmissionAddress.value()].tx_used_size = txBufferIndex;
+        return transmitted;
+    }   
+    return 0;
+}
+
+size_t TwoWire::write(const uint8_t* data, size_t quantity)
+{
+    if (transmissionAddress)
+    {
+        for(size_t i = 0; i < quantity; ++i){
+            write(data[i]);
+        }
```

Comment:
You really want to wait an eternity; this should be a single proto-req and a memcpy into the buffer